File: 贪心/买卖股票1/cpp/main.cpp
#include "bits/stdc++.h"
using namespace std;
//https://leetcode.cn/problems/best-time-to-buy-and-sell-stock/description/
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min_price = INT32_MAX;
        int max_profit = 0;
        for(int& price:prices){
            min_price = std::min(min_price,price);
            max_profit = std::max(max_profit,price-min_price);
        }
        return max_profit;
    }
};
int main(){
    Solution s;
    return 0;
}

File: 贪心/买卖股票1/java/Main.java
import java.io.*;
// 1:无需package
// 2: 类名必须Main, 不可修改

public class Main {

    public int maxProfit(int[] prices) {
        int minPrice=Integer.MAX_VALUE;
        int maxProfit=0;
        for(int x:prices){
            minPrice=Math.min(minPrice,x);
            maxProfit=Math.max(maxProfit,x-minPrice);
        }
        return maxProfit;
    }

    public static void main(String[] args) throws IOException {
        Main main = new Main();
    }
}